### DISCUSSION TITLE
WIKI corrections/submissions

### DISCUSSION THREAD

**Discussion by atomGit** [evilsocket/opensnitch#1115] [General]
* `wiki/Getting-started > Some of these processes are:`
`/usr/bin/kdeinit5` <- file doesn't exist (KDE Frameworks Version: 5.115.0)

* `wiki/Process-monitor-dialog` - there is no 'processes' tab - perhaps it's been removed? or not available on some systems?

* `wiki/Configurations`

> There're some scenarios where this option is useful/needed though, for example when connecting to VPNs, mount NFS shares or intercepting forwarded connections from containers.

seems the VPN part can be edited to reflect if ebpf is the process monitor?

* `wiki/Rules#best-practices > Disable unprivileged namespace`

my understanding is that the advice here (set 0) is a very double-edge sword because, at 0, applications like a web browser will not be able to sandbox

ref: [Default enabled unprivileged_userns_clone -Security - Chat about Zorin - Zorin Forum](https://forum.zorin.com/t/default-enabled-unprivileged-userns-clone-security/30609)

* `wiki/Rules-examples > Prioritize rules`

> Rules are checked in alphabetical order, ...
> The second step is to check the box [x] Priority of a rule.
>
> Rules with Action: Deny or rules with the box Priority checked will take precedence over the others.

i think this needs to be explained better? what's the diff between a 000 rule and, for example, a 000 rule with priority enabled?

re: wiki submissions: sorry, but i don't wish to fork the repo and do pulls (or have anything more to do with github than necessary due to f'n M$ ownership)

**Comment by gustavo-iniguez-goya** [comment]
Hi @atomGit !

Thank you for reviewing the wiki and the suggestions! Many parts of the wiki haven't been updated for quite some time unfortunately.

> /usr/bin/kdeinit5 <- file doesn't exist (KDE Frameworks Version: 5.115.0)

What's the equivalent on Plasma 6?

> wiki/Process-monitor-dialog - there is no 'processes' tab - perhaps it's been removed? or not available on some systems?

Updated. Many screenshots are still from version v1.0.0.

> my understanding is that the advice here (set 0) is a very double-edge sword because, at 0, applications like a web browser will not be able to sandbox

Yeah, the problem is that it can be used to bypass the rules. With the new feature for checking processes' checksums fn the development branch, this is not a problem anymore, but for the stable v1.6.x it is.

I added it to the wiki primarily to make users aware of the issue.

>  think this needs to be explained better? what's the diff between a 000 rule and, for example, a 000 rule with priority enabled?

I've also updated that section.

Please, if you have more suggestions, write the correction here and I'll update the wiki for you. Thank you!!

**Comment by atomGit** [comment]
> What's the equivalent on Plasma 6 [kdeinit5]?

dunno - i'm on 5.x

> Yeah, the problem is that it can be used to bypass the rules. With the new feature for checking processes' checksums fn the development branch, this is not a problem anymore, but for the stable v1.6.x it is.

mmm, i see - what's worse, an un-sandboxed browser or firewall trickery? i guess that's anyone's guess :)

> Please, if you have more suggestions ...

sure thing! and thanks for handling it

**Comment by atomGit** [comment]
re: `wiki/Process-monitor-dialog`

ok, i see what happened now - i said "there is no 'processes' tab ...", meaning that process _info_ was missing altogether

the process info is there, but the "node" column was so wide (by default i believe) that everything else was pushed off-screen and i didn't immediately notice the scroll bar :)

re: `wiki/Rules-examples > Prioritize rules`

the explanation for the priority switch technically makes sense on that page but the info in `wiki/Rules#some-considerations`  explains it in a crystal clear manner ...

> if the connection matches this rule, it'll allow this rule and **won't continue evaluating the rest of rules**.

consider adding that to the `wiki/Rules-examples > Prioritize rules` page, maybe something like ...

"If the connection matches a Priority rule, the firewall will allow it and won't continue evaluating any further rules, even if they pertain to the same command/executable."

**Comment by atomGit** [comment]
* `wiki/Installation`

consider adding info for the Arch family ???

proposed info ...

the line "Since version > v1.5.2, opensnitch is available on Debian Bookworm 12 and ubuntu 23.04." would have to be edited to include Arch

**PACMAN**

```
$ sudo pacman -Syu
$ sudo pacman -S opensnitch
```

If not installed, also install the version of opensnitch-ebpf-module package that corresponds to the version of opensnitch. 

If you're running Manjaro Linux and opensnitch-ebpf-module fails to build, edit the build file (ref: [https://aur.archlinux.org/packages/opensnitch-ebpf-module](https://aur.archlinux.org/packages/opensnitch-ebpf-module)):

change: `KDIR="/usr/src/linux"`
to: `KDIR="/usr/lib/modules/$(uname -r)/build"`

Start opensnitch:
```
$ sudo systemctl enable --now opensnitch
$ sudo systemctl start opensnitch
```

If "opensnitch" doesn't work, try "opensnitchd" (required for Manjaro and possibly other Arch derivatives)
